lib/ruby_gc_logger.rb

```diff
@@ -0,0 +1,71 @@
+require 'singleton'
+class RubyGCLogger
+  include Singleton
+
+  def start_gc_statistics_thread(seconds = 60)
+    require 'objspace'
+    require 'miq-process'
+
+    # Make a time based filename
+    csv      = File.open(Rails.root.join("log", "#{Process.pid}.csv"), "w+")
+    csv.sync = true
+    # Thread.abort_on_exception = true
+    @gc_stat_thread = Thread.new do
+      csv.puts(gc_stat_header.join(",".freeze))
+      loop do
+        csv.puts(gc_stat_line.join(",".freeze))
+        sleep seconds
+      end
+    end
+
+    at_exit { csv.close }
+    csv.path
+  end
+
+  def self.gc_stat_now
+    instance.gc_stat_now
+  end
+
+  def gc_stat_now
+    @gc_stat_thread.wakeup
+    Thread.pass
+  end
+
+  private
+
+  def gc_stat_line
+    gc_stat       = GC.stat
+    live_objects  = gc_stat[:total_allocated_objects] - gc_stat[:total_freed_objects]
+    young_objects = live_objects - gc_stat[:old_objects]
+
+    [Time.now.iso8601, live_objects, young_objects, ObjectSpace.memsize_of_all] +
+      MiqProcess.processInfo.values_at(*miq_process_keys) +
```

Comment:
Need to remove the dependence on MiqProcess and instead use the standalone gem sys-proctable.